hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemListStatusIterator.java

```diff
@@ -0,0 +1,190 @@
+    super();
+  }
+
+  @Test
+  public void testListPath() throws Exception {
+    final AzureBlobFileSystem fs = getFileSystem();
+    String rootPath = "testRoot1";
+    final List<String> fileNames = createFiles(TEST_FILES_NUMBER, rootPath,
+        "testListPath");
+    AzureBlobFileSystemStore abfsStore = getAbfsStore(fs);
+    abfsStore.getAbfsConfiguration().setListMaxResults(10);
+    RemoteIterator<FileStatus> fsIt = fs.listStatusIterator(new Path(rootPath));
+    int itrCount = 0;
+    while (fsIt.hasNext()) {
+      FileStatus fileStatus = fsIt.next();
+      String pathStr = fileStatus.getPath().toString();
+      fileNames.remove(pathStr);
+      itrCount++;
+    }
+    assertEquals(TEST_FILES_NUMBER, itrCount);
+    assertEquals(0, fileNames.size());
+  }
+
+  @Test
+  public void testNextWhenNoMoreElementsPresent() throws Exception {
+    final AzureBlobFileSystem fs = getFileSystem();
+    String rootPathStr = "testRoot2";
+    Path rootPath = new Path(rootPathStr);
+    getFileSystem().mkdirs(rootPath);
+    RemoteIterator<FileStatus> fsItr = fs.listStatusIterator(rootPath);
+    fsItr = Mockito.spy(fsItr);
+    Mockito.doReturn(false).when(fsItr).hasNext();
+
+    RemoteIterator<FileStatus> finalFsItr = fsItr;
+    Assertions.assertThatThrownBy(() -> finalFsItr.next()).describedAs(
+        "next() should throw NoSuchElementException if hasNext() return "
+            + "false").isInstanceOf(NoSuchElementException.class);
+  }
+
+  @Test
+  public void testHasNextForEmptyDir() throws Exception {
+    final AzureBlobFileSystem fs = getFileSystem();
+    String rootPathStr = "testRoot3";
+    Path rootPath = new Path(rootPathStr);
+    getFileSystem().mkdirs(rootPath);
+    RemoteIterator<FileStatus> fsItr = fs.listStatusIterator(rootPath);
+    Assertions.assertThat(fsItr.hasNext())
+        .describedAs("hasNext returns false for empty directory").isFalse();
+  }
+
+  @Test
+  public void testHasNextForFile() throws Exception {
+    final AzureBlobFileSystem fs = getFileSystem();
+    String rootPathStr = "testRoot4";
+    Path rootPath = new Path(rootPathStr);
+    getFileSystem().create(rootPath);
+    RemoteIterator<FileStatus> fsItr = fs.listStatusIterator(rootPath);
+    Assertions.assertThat(fsItr.hasNext())
+        .describedAs("hasNext returns true for file").isTrue();
+  }
+
+  @Test
+  public void testHasNextForIOException() throws Exception {
+    final AzureBlobFileSystem fs = getFileSystem();
+    String rootPathStr = "testRoot5";
+    Path rootPath = new Path(rootPathStr);
+    getFileSystem().mkdirs(rootPath);
+    ListStatusRemoteIterator fsItr = (ListStatusRemoteIterator) fs
+        .listStatusIterator(rootPath);
+    Thread.sleep(1000);
+
+    String exceptionMessage = "test exception";
+    setPrivateField(fsItr, ListStatusRemoteIterator.class, "ioException",
+        new IOException(exceptionMessage));
+    setPrivateFinalField(fsItr, ListStatusRemoteIterator.class,
```

Comment:
Instead of testing failure case by setting ioException field, try to mock abfsStore or lower to give a failure response.

Reply:
Done